static/app/components/hovercard.tsx

```diff
@@ -149,91 +119,160 @@ class Hovercard extends React.Component<Props, State> {
         enabled: true,
         boundariesElement: 'viewport',
       },
-      ...(modifiers || {}),
+      ...(props.modifiers || {}),
     };
+    return modifiers;
+  }, [props.modifiers]);
 
-    const visible = show !== undefined ? show : this.state.visible;
-    const hoverProps =
-      show !== undefined
-        ? {}
-        : {onMouseEnter: this.handleToggleOn, onMouseLeave: this.handleToggleOff};
-
-    return (
-      <Manager>
-        <Reference>
-          {({ref}) => (
-            <span
-              ref={ref}
-              aria-describedby={this.tooltipId}
-              className={containerClassName}
-              {...hoverProps}
-            >
-              {this.props.children}
-            </span>
-          )}
-        </Reference>
-        {visible &&
-          (header || body) &&
-          ReactDOM.createPortal(
-            <Popper placement={position} modifiers={popperModifiers}>
-              {({ref, style, placement, arrowProps, scheduleUpdate}) => {
-                this.scheduleUpdate = scheduleUpdate;
-                return (
-                  <StyledHovercard
-                    id={this.tooltipId}
-                    visible={visible}
-                    ref={ref}
-                    style={style}
-                    placement={placement as Direction}
-                    offset={offset}
-                    className={cx}
-                    {...hoverProps}
-                  >
-                    {header && <Header>{header}</Header>}
-                    {body && <Body className={bodyClassName}>{body}</Body>}
-                    <HovercardArrow
-                      ref={arrowProps.ref}
-                      style={arrowProps.style}
-                      placement={placement as Direction}
-                      tipColor={tipColor}
-                      tipBorderColor={tipBorderColor}
-                    />
-                  </StyledHovercard>
-                );
-              }}
-            </Popper>,
-            this.portalEl
-          )}
-      </Manager>
-    );
-  }
+  const isVisible = React.useMemo(() => {
```

Comment:
do we need `useMemo` here?

Reply:
Probably not, I used it mostly because the value is computed, but it's not 100% necessary. In general, I still like to use useMemo for these types of things and just avoid executing the code

Reply:
I believe this comes with a cost and we should use it when really necessary, but maybe let's discuss this more in the next TSC.. what do you think?

Reply:
@priscilawebdev what kind of cost do you have in mind?

If we compare 
```ts
const isVisible = typeof show === "undefined" ? visible : props.show
```
to
```ts
const isVisible = React.useMemo(() => {
  // If show is not set, then visibility state is uncontrolled
  if(typeof show === "undefined"){
    return visible
  }
  return props.show
},[props.show, visible])
```

The inline version is short, but crammed into a single expression with a ternary operator (which is probably fine for now and could be fine forever if we dont expect it to change) and the code will be re-evaluated everytime Hovercard rerenderers.

For #2, the code is obviously a bit longer, but in my opinion more readable + our comment is closer to the line that it actually applies to and logic is only re-evaluated if either visible state changes or different value is passed for props.show. The con is that without exhaustive deps eslint rule, it's easy to forget updating your dependency and end up with stale isVisible data, but we can avoid that in the future.

I think this is a somewhat subjective line that we'll probably have to define as we grow the codebase and this example here is very simple which is why I think either of the solutions is ok, as long as we all agree to follow it. 

That said, had this been slightly different and if we computed some expensive value here, then I probably would have opted for always using useMemo to avoid unnecessarily re-running the computation.

Reply:
I see... but in this case, I think it's fine to re-evaluated the code always when the Hovercard (re)renders because it's not expensive.

Your point is valid but I would be careful when using `useMemo` and `useCallback`. 

My opinion about these hooks changed after reading this blog post https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
Yeah, 100% agree with that post. I dont think we care about performance in this case and we dont care about references here either, the code is too simple to bother. It's pretty much a question if we want to cram everything in a single expression or not, I dont mind either one - I'll inline it and remove the memo :)